core/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -523,11 +532,12 @@ public void failEngine(String reason, Throwable failure) {
                     // this must happen first otherwise we might try to reallocate so quickly
                     // on the same node that we don't see the corrupted marker file when
                     // the shard is initializing
-                    if (Lucene.isCorruptionException(failure)) {
+                    if (markAsCorrupted) {
                         try {
-                            store.markStoreCorrupted(ExceptionsHelper.unwrapCorruption(failure));
+                            store.markStoreCorrupted(unwrapCorruption("failed engine (reason: [" + reason + "])", failure));
```

Comment:
I think we can skip the unwrapCorruption and always use the IOException which captures the reason as well.